miden-tx/src/host/account_procs.rs

```diff
@@ -10,67 +13,74 @@ use crate::errors::TransactionHostError;
 // ================================================================================================
 
 /// A map of proc_root |-> proc_index for all known procedures of an account interface.
-pub struct AccountProcedureIndexMap(BTreeMap<Digest, u8>);
+pub struct AccountProcedureIndexMap(BTreeMap<Digest, BTreeMap<Digest, u8>>);
 
 impl AccountProcedureIndexMap {
     /// Returns a new [AccountProcedureIndexMap] instantiated with account procedures present in
     /// the provided advice provider.
     pub fn new(
-        account_code_commitment: Digest,
+        native_account_code_commitment: Digest,
         adv_provider: &impl AdviceProvider,
+        foreign_account_code_commitments: &[Digest],
     ) -> Result<Self, TransactionHostError> {
```

Comment:
nit: I wonder if we can combine native code comment and foreign code comments into a single parameter. The procedure map doesn't really care which one is which, right? Maybe something like:
```rust
pub fn new(
    account_code_commitment: Vec<Digest>,
    adv_provider: &impl AdviceProvider,
) -> Result<Self, TransactionHostError>
```